### ISSUE TITLE
Measure Name showing different in PowerBI

### ISSUE THREAD

**Issue by xushuo1987** [otykier/TabularEditor#533]
I used Tabular Editor to open my PBI model and did some measure duplication and modification. Then something weird happened.

My renamed measure is showing blue in Tabular Editor like this
![image](https://user-images.githubusercontent.com/68398773/87713670-9c451f00-c778-11ea-9634-6ad8278a2be9.png)

and those blue measures are showing incorrectly in PBI Desktop:
![image](https://user-images.githubusercontent.com/68398773/87713709-a9faa480-c778-11ea-8afb-51769851b137.png)

However, when I open the DAX formula of the PBI Measures, they are showing completely different information:
![image](https://user-images.githubusercontent.com/68398773/87713875-e928f580-c778-11ea-9a8f-7e783d3f686b.png)

As you can see from the above picture, I'm selecting a measure called "PY Written Cost MTD" but in the formula bar, it's called "Delivered Cost MTD PY". 

What could be the issue here? Can anyone suggest it?

Thanks

**Comment by xushuo1987** [comment]
To add a comment on this, I personally believe it's due to the translation but I never enable any translation by myself. How should I remove it?

Thanks

**Comment by brinew27** [comment]
I am having this same issue. When a measure it duplicated, the duplicate measure can be renamed in Tabular Editor (it shows up blue). The measure appears with the correct name in the DAX (both in Tabular Editor and Power BI Desktop), but the measure is listed with the original name in Power BI desktop, and even using the UI to rename it does nothing to the displayed name.

**Comment by brinew27** [comment]
Interesting. 

> To add a comment on this, I personally believe it's due to the translation but I never enable any translation by myself. How should I remove it?
> 
> Thanks

Interesting, your comment about translation led me to I think fix it. By default Tabular Editor is set to "(no translation)", so when you're renaming the measures, I believe it's not affecting the English translation, which is probably the default we see in Power BI. When I changed the translation to English, the measures again showed the original measure name in Tabular Editor. When I renamed them in Tabular editor to match the DAX, the measure no longer showed up Blue and when I saved it, they updated correctly in power BI.

![Translation](https://user-images.githubusercontent.com/872698/87829101-2f428f80-c833-11ea-9cb6-6d0e2c0ca33d.jpg)

**Comment by otykier** [comment]
Here's some more details on what's going on:

By default, Power BI Desktop adds a default culture to the model based on the OS language - in your case, it's "en-US" (it appears under Model > Translations:

![image](https://user-images.githubusercontent.com/8976200/87938748-526d7900-ca97-11ea-84e2-e3e1799714b6.png)

Wben you rename an existing measure in Tabular Editor, the tool will provide a default translation that matches the new name, for every culture in the model, where the translated name is either blank or identical to the objects original name. You can view the translated name(s) in the properties grid as shown in the screenshot here:

![image](https://user-images.githubusercontent.com/8976200/87939459-6665aa80-ca98-11ea-90c8-ed4d553cfbd1.png)

The object remains black as long as an object does not have any non-blank translated names that are different from the name itself. The issue occurs when you use the "Duplicate measure" or copy/paste a measure in your model. By default, this operation copies all translations on that object as well, but the object name is changed since you can't have two or more measures with the same name. Thus, the translated name is no longer aligned with the object name, so the next time you rename the object, the translated name is left as-is. At this point, the object turns blue to indicate that the object has one or more translations that would change the appearance of the object in client tools:

![image](https://user-images.githubusercontent.com/8976200/87939969-218e4380-ca99-11ea-8077-8096f78068da.png)

Selecting the culture in the dropdown at the top of the screen lets you preview what the objects are going to look like in client tools using that specific culture:

![image](https://user-images.githubusercontent.com/8976200/87940056-4682b680-ca99-11ea-8579-052ef5f45c5c.png)

So to fix this issue, you have a couple of options:

- Work your way through the objects and set the translated name to blank or make it identical to the object name. This operation can be scripted (see below).
- Delete the culture from the model. This removes all translations for that culture at once. This is my recommended solution if you don't need the metadata translations in your model.
- To avoid the disconnected translations in the first place when copy/pasting measures, remove the checkmark from "Include translations" in Tabular Editor's File > Preferences > Clipboard Operations setting.

To reset translated names to the object names across all your model measures, use the following script (paste into Tabular Editor's "Advanced Scripting" tab and hit F5):

```csharp
foreach(var m in Model.AllMeasures)
    m.TranslatedNames.Reset();
```

**Comment by otykier** [comment]
Update: This will be fixed in 2.11.7, so that translations are not included in copy/paste or duplicate operations by default, to avoid further confusion. But anyone who installed an earlier version of Tabular Editor will need to go into File > Preferences and remove the checkmark from "Include translations".

**Comment by otykier** [comment]
Fixed in [2.11.7](https://github.com/otykier/TabularEditor/releases/latest).
